draft-reddy-ipsecme-ikev2-pqc-auth.md

```diff
@@ -137,9 +130,15 @@ The following combinations are defined in SLH-DSA [FIPS205]:
 
-IKEv2 can use arbitrary signature algorithms as described in {{!RFC7427}}. The "Digital Signature" authentication method, as defined in {{!RFC7427}}, supersedes previously defined signature authentication methods. In this case, the different combinations of SLH-DSA are identified via AlgorithmIdentifier ASN.1 objects, as specified in {{I-D.ietf-lamps-cms-sphincs-plus}}. The final version of SLH-DSA [FIPS205] is expected to define two signature modes: pure mode and predigest mode.  This document only specifies the use of predigest mode for signature-based authentication in IKEv2. In case of predigest mode, SLH-DSA internally performs randomized message compression using a keyed hash function that can process arbitrary length messages. The data signed to prove the identity of the initiator and responder (as discussed in Section 2.15 of {{!RFC7427}}) typically fits within the memory constraints of the devices involved in the IKEv2 exchange. The data consists of nonces, SPIs, and the initial exchange messages, which are manageable in size. In order to signal within IKE that no pre-hashing needs to be done with SLH-DSA, the "Identity" (5) value defined in {{!RFC8420}} MUST be set in the SIGNATURE_HASH_ALGORITHMS notification to indicate that pre-hashing is not required. 
+{{!RFC7427}} defines the SIGNATURE_HASH_ALGORITHMS notification, where each side of the IKE negotiation lists its supported hash algorithms, assuming that signature schemes involve a hashing phase followed by a signature phase. The data signed to prove the identity of the initiator and responder (as discussed in Section 2.15 of {{!RFC7427}}) typically fits within the memory constraints of the devices involved in the IKEv2 exchange, consisting of nonces, SPIs, and the initial exchange messages, which are manageable in size. To signal that no pre-hashing is required for either ML-DSA or SLH-DSA at the IKEv2 level, the 'Identity' (5) value defined in {{!RFC8420}} can be set in the SIGNATURE_HASH_ALGORITHMS notification. When hashing is applied, the digest must be generated using an approved hash function or XOF (e.g., from [FIPS180] or [FIPS202]) that provides at least 𝜆 bits of classical security strength against collision and second preimage attacks, as per Table 4 of [FIPS202].
```

Comment:
I'm a bit confused. Previously we state that only pure signature mode is specified for IKEv2, thus the only acceptable hash algorithm to appear in SIGNATURE_HASH_ALGORITHS is Identity (thus it should probably be "MUST be set" and not "can be set" there). But later the text says that hashing may be applied. Am I missing something? Perhaps this is about hashing performed "inside" signature algorithm? Then isn't it a signature algorithm's matter and not an IKEv2 matter?

Reply:
In the previous version of the draft, we opted for the pre-digest mode. In this mode, hashing is not required at the IKEv2 level. However, the LAMPS WG recently discussed not to use pre-hash mode, so I updated the draft to use pure mode. In pure mode, we have two options: we can either perform hashing at the IKEv2 level or avoid it altogether since the data to be signed is not large.

Reply:
Perhaps this should be further clarified in the draft with some guidance when to use hashing at IKEv2 level: I suppose this may be required if the size of the IKE_SA_INIT messgae is too large, (with some definition of "too") - this may happen if Large Payload from Yoav draft is adopted and TCP is used.

Reply:
A message size of 4 KB seems like a reasonable 'too large' threshold when dealing with HSM-based signing, though the limit may be lower for constrained IoT devices. To avoid confusion and ensure consistent performance, I will update the draft to mandate pre-hashing at the IKEv2 level.